app/src/main/java/com/example/lexis/fragments/FeedFragment.java

```diff
@@ -2,65 +2,104 @@
-     * this fragment using the provided parameters.
-     *
-     * @param param1 Parameter 1.
-     * @param param2 Parameter 2.
-     * @return A new instance of fragment FeedFragment.
-     */
-    // TODO: Rename and change types and number of parameters
-    public static FeedFragment newInstance(String param1, String param2) {
-        FeedFragment fragment = new FeedFragment();
-        Bundle args = new Bundle();
-        args.putString(ARG_PARAM1, param1);
-        args.putString(ARG_PARAM2, param2);
-        fragment.setArguments(args);
-        return fragment;
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        binding = FragmentFeedBinding.inflate(inflater);
+        return binding.getRoot();
     }
 
     @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        if (getArguments() != null) {
-            mParam1 = getArguments().getString(ARG_PARAM1);
-            mParam2 = getArguments().getString(ARG_PARAM2);
-        }
+    public void onViewCreated(@NonNull @NotNull View view, @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        articles = new ArrayList<>();
+        fetchWikipediaArticle("Elon Musk");
+        
+        binding.btnArticleFragment.setOnClickListener(v -> {
+            final FragmentManager fragmentManager = getActivity().getSupportFragmentManager();
+            final Fragment articleFragment = ArticleFragment.newInstance(articles.get(0));
+
+            fragmentManager.beginTransaction()
+                    .replace(R.id.fragmentContainer, articleFragment)
+                    .addToBackStack("")
+                    .commit();
+        });
     }
 
-    @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_feed, container, false);
+    private void fetchWikipediaArticle(String title) {
+        AsyncHttpClient client = new AsyncHttpClient();
+        RequestParams params = new RequestParams();
+        params.put("action", "query");
+        params.put("format", "json");
+        params.put("titles", title);
+        params.put("prop", "extracts");
+        params.put("explaintext", true);
+        params.put("exintro", true);
+
+        client.get(WIKI_ARTICLE_URL, params, new JsonHttpResponseHandler() {
+            @Override
+            public void onSuccess(int statusCode, Headers headers, JSON json) {
+                JSONObject jsonObject = json.jsonObject;
+                try {
+                    JSONObject query = jsonObject.getJSONObject("query");
+                    JSONObject pages = query.getJSONObject("pages");
```

Comment:
Will any of the fields in these objects be nullable?

Reply:
If not, you can try chaining this:

```
JSONObject pages = jsonObject
    .getJSONObject("query")
    .getJSONObject("pages");
```

If it's nullable for whatever reason, this and the above will break your code without nullchecks.